packages/admin/src/table/filterbar/FilterBarPopOverFormField.tsx

```diff
@@ -0,0 +1,187 @@
+import { Box, Button, ButtonProps, Popover, Typography } from "@material-ui/core";
+import { makeStyles, Theme, ThemeOptions, useTheme } from "@material-ui/core/styles";
+
+import get = require("lodash.get");
+import * as React from "react";
+import { FieldRenderProps, Form, useForm } from "react-final-form";
+
+import { IFilterBarField } from "./FilterBar";
+import { FilterBarDirtyFieldsBadge } from "./FilterBarDirtyFieldsBadge";
+
+export interface PopoverFormFieldThemeProps {
+    submitButton?: ButtonProps;
+    resetButton?: ButtonProps;
+}
+
+export type CometAdminFilterBarPopOverFormFieldClassKeys =
+    | "root"
+    | "styledBox"
+    | "labelWrapper"
+    | "popoverContentContainer"
+    | "buttonsContainer"
+    | "submitContainer"
+    | "resetCloseContainer";
+
+const useStyles = makeStyles(
+    (theme: Theme) => ({
+        root: {
+            position: "relative",
+        },
+        styledBox: {
+            position: "relative",
+            alignItems: "center",
+            padding: "10px 20px",
+            cursor: "pointer",
+            display: "flex",
+
+            "&:after": {
+                borderTop: `4px solid ${theme.palette.grey[300]}`,
+                borderRight: "4px solid transparent",
+                borderLeft: "4px solid transparent",
+                position: "absolute",
+                display: "block",
+                right: "10px",
+                content: "''",
+                top: "50%",
+                height: 0,
+                width: 0,
+            },
+        },
+        labelWrapper: {
+            marginRight: "15px",
+            boxSizing: "border-box",
+        },
+        popoverContentContainer: {
+            "& [class*='CometAdminFormFieldContainer-root']": {
+                boxSizing: "border-box",
+                padding: "20px",
+                marginBottom: 0,
+            },
+        },
+        buttonsContainer: {
+            borderTop: `1px solid ${theme.palette.grey[300]}`,
+            justifyContent: "space-between",
+            padding: "10px 15px",
+            display: "flex",
+        },
+        submitContainer: {},
+        resetContainer: {
+            marginRight: "15px",
+        },
+    }),
+    { name: "CometAdminFilterBarPopOverFormField" },
+);
+
+interface IFormFieldProps extends FieldRenderProps<any> {
+    field: IFilterBarField;
+    handleSubmit: () => void;
+}
+
+export const FilterBarPopOverFormField: React.FunctionComponent<IFormFieldProps & PopoverFormFieldThemeProps> = ({ field, dirtyFieldsBadge }) => {
+    const DirtyFieldsBadgeComponent = field.dirtyFieldsBadge ? field.dirtyFieldsBadge : FilterBarDirtyFieldsBadge;
```

Comment:
What's the motivation behind this? Is there a real use case where I'd like to override the badge? 🤔

Reply:
i added this due to the fact that maybe someone needs a custom calculation for the counted number of the form :) this was the easiest way to cover all possible ways to count the dirty fields :) so i added a default dirtyField Badge which covers the basic count of dirty fields, and for custom count you can override it with your own component :)

Reply:
The term "dirty fields" refers to fields which have been changed but their value has not been submitted yet.  This badge is actually useful is for counting how many "filters are active". So probably `activeFilterBadge` or `activeFilterCountBadge` might be a better name? The fact that the badge uses dirty fields to count the the active filters is an implementation detail.

Reply:
i renamed it now to activeFilterBadge